inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -685,12 +685,93 @@ layout layout_optimizer::get_expected_layout(layout const& current_layout,
+        } else if ((_optimization_attributes.bs_fs_yx_bsv16_fsv16_network && expected_tensor.batch[0] % 16 == 0 &&
+            convolution_bs_fs_yx_bsv16_fsv16_opt(input_layout, output_layout, weights_layout, prim)) && is_2d) {
+            expected_format = cldnn::format::bs_fs_yx_bsv16_fsv16;
+        } else if ((_optimization_attributes.b_fs_yx_fsv16_network &&
+            convolution_b_fs_yx_fsv16_opt(input_layout, output_layout, weights_layout, prim)) && is_2d) {
+            if (is_dw)
+                expected_format = cldnn::format::b_fs_yx_fsv32;
+            else
+                expected_format = cldnn::format::b_fs_yx_fsv16;
+        } else if ((_optimization_attributes.b_fs_zyx_fsv16_network &&
+            convolution_b_fs_zyx_fsv16_opt(input_layout, output_layout, weights_layout, prim))) {
+            expected_format = cldnn::format::b_fs_zyx_fsv16;
+        } else {
+            expected_format = imad_case(node);
+        }
+        expected_tensor = current_layout.size;
+    } else if (use_onednn_impls && node.input().get_output_layout().data_type == data_types::f16 && is_2d) {
+        expected_tensor = current_layout.size;
+
+        if (input_layout.size.batch[0] >= 16 && valid_post_ops) {
+            if (node.get_output_layout().data_type == node.input().get_output_layout().data_type) {
+                if (non_grouped || valid_grouped || is_dw) {
+                    expected_format = cldnn::format::bs_fs_yx_bsv32_fsv16;
+                } else {
+                    expected_format = cldnn::format::b_fs_yx_fsv16;
+                }
+            } else {
+                expected_format = cldnn::format::bs_fs_yx_bsv16_fsv16;
+            }
+        } else {
+            expected_format = cldnn::format::b_fs_yx_fsv16;
+        }
+    } else if (use_onednn_impls && node.input().get_output_layout().data_type == data_types::f16 &&
+               convolution_bs_fs_yx_bsv16_fsv16_opt(node.input().get_output_layout(), output_layout, weights_layout, prim) &&
+               (node.get_output_layout().data_type == node.input().get_output_layout().data_type ||
+               !data_type_traits::is_floating_point(node.input().get_output_layout().data_type))) {
+        expected_tensor = current_layout.size;
+        if (prim->groups == 1 || (output_layout.size.feature[0] % 16 == 0 && node.input().get_output_layout().size.feature[0] % 16 == 0)) {
+            expected_format = cldnn::format::bs_fs_yx_bsv32_fsv16;
+        } else {
+            expected_format = cldnn::format::bs_fs_yx_bsv16_fsv16;
+        }
+
+    /* ***************************************************************************************** */
+    /* *************************** Native impls layout selection part ************************** */
+    /* ***************************************************************************************** */
+
+    } else if ((input_layout.data_type == data_types::u8 || input_layout.data_type == data_types::i8)) {
```

Comment:
You duplicate this check, I think it can be checked before `if - else` branching as a separate bool variable.

Reply:
Fixed